Third_HW/HW_3_fixed.js

```diff
@@ -0,0 +1,90 @@
+//Задача 1
+
+console.log(addThemAll(2,4)); // 6
+console.log(addThemAll(1,2,3,4)); // 10
+console.log(addThemAll(5,5,10)); // 20
+
+function addThemAll(...args) {
+    let sum = 0;
+    for(let item of args) {
+        sum += item; 
+    }
+    return sum;
+}
+
+//Задача 2
+
+console.log(multiply(5)(5))	// 25
+console.log(multiply(2)(-2))	// -4
+console.log(multiply(4)(3))		// 12
+
+function multiply(a) {
+    return function(b) {
+    return a * b;
+    }
+
+}
+
+//Задача 3
+
+const movies = [
+	{
+		movieName: 'The Thing',
+		releaseYear: 1982,
+		directedBy: 'Carpenter',
+		runningTimeInMinutes: 109,
+	},
+	{
+		movieName: 'Aliens',
+		releaseYear: 1986,
+		directedBy: 'Cameron',
+		runningTimeInMinutes: 137,
+	},
+	{
+		movieName: 'Men in Black',
+		releaseYear: 1997,
+		directedBy: 'Sonnenfeld',
+		runningTimeInMinutes: 98,
+	},
+	{
+		movieName: 'Predator',
+		releaseYear: 1987,
+		directedBy: 'McTiernan',
+		runningTimeInMinutes: 107,
+	},
+];
+
+//console.log(movies.sort(byProperty('releaseYear', '>'))); 
+// виведе масив фільмів посортованих по року випуску, від старішого до новішого
+//console.log(movies.sort(byProperty('runningTimeInMinutes', '<'))); 
+// виведе масив фільмів посортованих по їх тривалості, від найдовшого до найкоротшого
+//console.log(movies.sort(byProperty('movieName', '>'))); 
+// виведе масив фільмів посортованих по назві, в алфавітному порядку
+
+function byProperty(property, direction) {
+return function (a, b) {
+	if (direction === '>'){
+		if (a[property] > b[property]) return 1;
+		if (a[property] < b[property]) return -1;
+	}
+	if (direction === '<'){
+		if (a[property] < b[property]) return 1;
+		if (a[property] > b[property]) return -1;
+	}
+
+	return 0;
+} 
+	
+};
+
+
+
+//Задача 4
+
+const userNames = ['Петро', 'Емма', 'Петро', 'Емма', 'Марта', 'Яна', 'Василь', 'Антон', 'Олена', 'Емма'];
+
+function filterUnique(array) {
+return Array.from([...new Set(userNames)]);
```

Comment:
Тут масив робиться з масиву. Треба або:
```suggestion
return [...new Set(userNames)];
```
або:
```suggestion
return Array.from(new Set(userNames));
```

Reply:
Результат обох варіантів однаковий, тільки механізм трохи інший